src/main/java/com/innohub/innosorter/authentication/AuthenticationService.java

```diff
@@ -1,26 +1,34 @@
 package com.innohub.innosorter.authentication;
 
+import com.innohub.innosorter.util.ApplicationConstants;
+
 public class AuthenticationService {
 
-	protected String emptyPasswordMessage = "Empty Password";
-	protected String nullPasswordMessage = "Null Password";
 	RegistrationServiceDao registrationServiceDao = new RegistrationServiceDao();
-	protected String unsuccessfulPasswordUpdateMessage = "Same as old password";
-	protected String successfulPasswordUpdateMessage = "Password has been updated";
-	protected String invalidUserMessage = "Invalid Username";
-	protected String invalidPasswordMessage = "Invalid Password";
 	
-	public String addNewUser(String username, String password) {
+	public String addNewUser(String username, String password, String userType) {
 		
+		if(username == null){
+			if(password == null){
+				throw new RuntimeException(ApplicationConstants.NULL_USERNAME_AND_PASSWORD_MSG);
+			}
+			throw new RuntimeException(ApplicationConstants.NULL_USERNAME_MSG);
+		}
+		if(username.isEmpty()){
+			if(password.isEmpty()){
+				throw new RuntimeException(ApplicationConstants.EMPTY_USERNAME_AND_PASSWORD_MSG);
+			}
+			throw new RuntimeException(ApplicationConstants.EMPTY_USERNAME_MSG);
+		}
 		if(password!= null){
 			//if the password is an empty string
 			if(password.isEmpty()){
-				throw new RuntimeException(emptyPasswordMessage);
+				throw new RuntimeException(ApplicationConstants.EMPTY_PASSWORD_MSG);
 			}
-			return registrationServiceDao.registerUser(username, password);
+			return registrationServiceDao.registerUser(username, password, userType);
 		}
 		else {
-			throw new RuntimeException(nullPasswordMessage);
+			throw new RuntimeException(ApplicationConstants.NULL_PASSWORD_MSG);
```

Comment:
Please consider we might not need to have too many nested conditions, maybe just 4 flat condition is enough